analyser/module_analyser_call.c2

```diff
@@ -183,7 +184,8 @@ fn QualType Analyser.analyseCallExpr(Analyser* ma, Expr** e_ptr) {
 
     u32 expected_args = func_num_args - num_auto_args - isTypeFuncCall;
     if (call_num_args < expected_args) {
-        ma.error(call.getEndLoc(), DiagTooFewArgs, fd.getDiagKind(), expected_args, call_num_args);
+        // error loc is closing parenthesis
+        ma.error(call.getEndLoc() - 1, DiagTooFewArgs, fd.getDiagKind(), expected_args, call_num_args);
```

Comment:
The diag was exactly where it was supposed to be, at the ')'. Now it's pointing at the last char of the last arg

Reply:
The semantics of `endLoc()` have changed: it is no longer the starting location of the end token (the ')'), it is now always the offset of the character **after** the token.  The same holds for `endLoc` local variables, I always store the location after a given token.  For consistency, `prev_loc` should be renamed `prev_endLoc`.

This simplifies the range calculations: `{ e.getStartLoc(), e.getEndLoc() }` is the exact source fragment for a given expression, and `diag` no longer needs `getTokenEnd()` to compute the range.

For the missing arguments, we indeed want to point to the `)`, hence `call.getEndLoc() - 1` as we can assume that the function call ends with a `)`.